### PR TITLE
SITL: allow to set sysid at start

### PR DESCRIPTION
**Pull request by khancyr**
This is really useful for all swarm application.
The main issue with sysid setting is that it is loaded early on vehicle initialization, changing this parameters need a reboot. This PR allow to pass --sysid X to SITL cmdline to set the sysid on start.
It also fix SPEEDUP parameter been set to X in SITL backend but at 1 in parameter if we use the -w flag. This issue comes to the fact that when we erase the EEPROM with -w, the vehicle will start by searching the FORMAT_VERSION parameter in the EEPROM, as we erase it, the vehicle won't find it and trigger another erase and load the default parameters. Thus every parameter pass on cmdline where lost ...This PR address this issue by allow to load the cmdline parameter in the same manner than the default file. That way the parameter are setted on start and are written back in case of erase.

On the SYSID matter. The current alternative where to launch the vehicle, change the SYSID_THISMAV parameter and reboot ... quite long.
Generate a parameters file with the SYSID_THISMAV modification and pass it with the --defaults option : example in https://github.com/ArduPilot/ardupilot/blob/master/libraries/SITL/examples/Morse/start_follow.sh

The option is also added to sim_vehicle.py that already have some option for swarm launch

### CODE REVIEW

**Review by tridge — APPROVED** [review]

File: libraries/AP_HAL_SITL/Util.cpp [hunk 1]
```diff
@@ -138,3 +139,10 @@ enum AP_HAL::Util::safety_state HALSITL::Util::safety_switch_state(void)
     }
     return sitl->safety_switch_state();
 }
+
+void HALSITL::Util::set_cmdline_parameters()
+{
+    for (auto param: sitlState->cmdline_param) {
+        AP_Param::set_and_save_by_name(param.name, param.value);
```
Comment (tridge) [inline libraries/AP_HAL_SITL/Util.cpp:146]: why not set_default_by_name()?
I don't think we want the command line options to persist to the next run, where the user may not provide the option at all

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_SITL/Util.cpp (hunk 1 above)
Comment (khancyr) [reply libraries/AP_HAL_SITL/Util.cpp]: done

**Comment by khancyr** [comment]
test done : 
` 4824  sim_vehicle.py -v ArduCopter -f hexa --sysid 42 -S 5 --map --console`
SYSID = 42
SPEEDUP = 5

` 4825  sim_vehicle.py -v ArduCopter -f hexa -w --map --console`
SYSID = 1
SPEEDUP = 1

` 4826  sim_vehicle.py -v ArduCopter -f hexa --sysid 42 -S 5 --map --console`
SYSID = 42
SPEEDUP = 5

` 4827  sim_vehicle.py -v ArduCopter -f hexa --sysid 42 -S 5 -w --map --console`
SYSID = 42
SPEEDUP = 5

` 4828  sim_vehicle.py -v ArduCopter -f hexa  --map --console`
SYSID = 1
SPEEDUP = 1

` 4829  sim_vehicle.py -v ArduCopter -f hexa --sysid 42 -S 5 -w --map --console`
SYSID = 42
SPEEDUP = 5

` 4830  sim_vehicle.py -v ArduCopter -f hexa --sysid 42 -w --map --console`
SYSID = 42
SPEEDUP = 1

` 4831  sim_vehicle.py -v ArduCopter -f hexa --sysid 42 -S 5 -w --map --console`
SYSID = 42
SPEEDUP = 5

` 4832  sim_vehicle.py -v ArduCopter -f hexa --sysid 42  --map --console`
SYSID = 42
SPEEDUP = 1

**Comment by tridge** [comment]
> test done :

thanks!

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
@khancyr on the dev call we realised that just using set() may do the right thing, and avoids both the persistence and existing value issues. It all depends on this being called at the right stage of starup so set() doesn't get overridden

**Comment by khancyr** [comment]
ok so I should change set_default_by_name by set_by_name. I have already test that and I shouldn't call it at the right place as it is always overrided by eeprom value

**Comment by tridge** [comment]
needs another rebase, sorry

**Comment by khancyr** [comment]
rebased

**Comment by tridge** [comment]
still failing rover CI ?

**Comment by khancyr** [comment]
there is no issues with CI, it could be merged !

**Review by tridge — APPROVED** [review]

**Pull request merged**